flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCTableSourceITCase.java

```diff
@@ -18,93 +18,167 @@
 
-import java.util.ArrayList;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
 import java.util.List;
 
+
 /**
- * IT case for {@link JDBCTableSource}.
+ * ITCase for {@link JDBCTableSource}.
  */
-public class JDBCTableSourceITCase extends JDBCTestBase {
-
-	private static final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
-	private static final EnvironmentSettings bsSettings = EnvironmentSettings.newInstance().useBlinkPlanner().inStreamingMode().build();
-	private static final StreamTableEnvironment tEnv = StreamTableEnvironment.create(env, bsSettings);
-
-	static final String TABLE_SOURCE_SQL = "CREATE TABLE books (" +
-		" id int, " +
-		" title varchar, " +
-		" author varchar, " +
-		" price double, " +
-		" qty int " +
-		") with (" +
-		" 'connector.type' = 'jdbc', " +
-		" 'connector.url' = 'jdbc:derby:memory:ebookshop', " +
-		" 'connector.table' = 'books', " +
-		" 'connector.driver' = 'org.apache.derby.jdbc.EmbeddedDriver' " +
-		")";
-
-	@BeforeClass
-	public static void createTable() {
-		tEnv.sqlUpdate(TABLE_SOURCE_SQL);
+public class JDBCTableSourceITCase extends AbstractTestBase {
+
+	public static final String DRIVER_CLASS = "org.apache.derby.jdbc.EmbeddedDriver";
+	public static final String DB_URL = "jdbc:derby:memory:test";
+	public static final String INPUT_TABLE = "jdbcSource";
+
+	@Before
+	public void before() throws ClassNotFoundException, SQLException {
+		System.setProperty("derby.stream.error.field", JDBCTestBase.class.getCanonicalName() + ".DEV_NULL");
+		Class.forName(DRIVER_CLASS);
+
+		try (
+			Connection conn = DriverManager.getConnection(DB_URL + ";create=true");
+			Statement statement = conn.createStatement()) {
+			statement.executeUpdate("CREATE TABLE " + INPUT_TABLE + " (" +
+					"id BIGINT NOT NULL," +
+					"timestamp6_col TIMESTAMP, " +
+					"timestamp9_col TIMESTAMP, " +
+					"time_col TIME, " +
+					"real_col FLOAT(23), " +    // A precision of 23 or less makes FLOAT equivalent to REAL.
+					"double_col FLOAT(24)," +   // A precision of 24 or greater makes FLOAT equivalent to DOUBLE PRECISION.
+					"decimal_col DECIMAL(10, 4))");
+			statement.executeUpdate("INSERT INTO " + INPUT_TABLE + " VALUES (" +
+					"1, TIMESTAMP('2020-01-01 15:35:00.123456'), TIMESTAMP('2020-01-01 15:35:00.123456789'), " +
+					"TIME('15:35:00'), 1.175E-37, 1.79769E+308, 100.1234)");
+			statement.executeUpdate("INSERT INTO " + INPUT_TABLE + " VALUES (" +
+					"2, TIMESTAMP('2020-01-01 15:36:01.123456'), TIMESTAMP('2020-01-01 15:36:01.123456789'), " +
+					"TIME('15:36:01'), -1.175E-37, -1.79769E+308, 101.1234)");
+		}
+	}
+
+	@After
+	public void clearOutputTable() throws Exception {
+		Class.forName(DRIVER_CLASS);
+		try (
+			Connection conn = DriverManager.getConnection(DB_URL);
+			Statement stat = conn.createStatement()) {
+			stat.execute("DROP TABLE " + INPUT_TABLE);
```

Comment:
Should we use `stat.executeUdpate`?

Reply:
execute can be used for any SQL statement, but executeUpdate is more clearly here. I will fix this.